crates/primitives/src/constants.rs

```diff
@@ -89,6 +89,9 @@ pub const EMPTY_TRANSACTIONS: H256 = EMPTY_SET_HASH;
 /// Withdrawals root of empty withdrawals set.
 pub const EMPTY_WITHDRAWALS: H256 = EMPTY_SET_HASH;
 
+/// The number of blocks to unwind during a reorg that already became a part of canonical chain.
+pub const REORG_UNWIND_DEPTH: u64 = 3;
```

Comment:
I'd like a bit more context here, why we're doing this.
and that using 3 blocks should reduce likelihood by a lot.
I guess this factor is essentially network dependent, perhaps we should even make this a field of the header stage? or not even include it in the `DetachedHead` and leave it up to the caller to decide what to do?